automation/release.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+function cleanup_gh_install() {
+    rm -f "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    rm -rf "gh_${GH_CLI_VERSION}_linux_amd64/"
+}
+
+function ensure_gh_cli_installed() {
+    if command -V gh; then
+        return
+    fi
+
+    trap 'cleanup_gh_install' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
```

Comment:
@dhiller I can't find where the ```GH_CLI_VERSION``` env variable is being defined. In case it is defined by the process that invokes this script, should we validate that it was indeed defined and exit if it's not defined?

Reply:
Here's the job definition where the script actually gets called: https://github.com/kubevirt/project-infra/pull/892/files#diff-baa3b19d79131de5e52e898eb3671fc4b665d327864aa84e3ccf1df16765864eR63

I was thinking about adding gh cli to the image but as we have only this use case currently, we decided to install it on the fly.

Reply:
BTW it will fail if it's not defined anyway as I set the -u flag on the script,